document agent connection+reconnection behaviors (#503)
ngrok/ngrok-docs: docs/agent/index.mdx

for some reason ngrok cannot restart, you may lose access to the agent
completely.

## Connectivity

> When the ngrok agent and Agent SDKs start, they establishes long-lived TLS

Reviewer: ```suggestion
When the ngrok agent and Agent SDKs start, they establish long-lived TLS
```

receive connections from ngrok's edge that are intended for your upstream
services.

### Address {#connectivity-address}

By default, the ngrok agent dials the following "agent ingress address" when it
connects to the ngrok service. This address resolves to a [dyanmic set of IP
Addresses](/network-edge/#ip-addresses).

```
connect.ngrok-agent.com:443
```

You can [customize the agent ingress address](/agent/ingress/) to brand it with
your own domain.

### DNS Resolution

When the ngrok agent dials the ngrok service to establish its TLS connections,
it resolves DNS for the [connection address](/agent/ingress/) which is defined
by the [`server_addr`](/agent/config/#server_addr) configuration property.
ngrok attempts to resolve DNS using multiple mechanisms so that it can
establish connectivity even in network environments where DNS service is
failing. ngrok attempts to resolve the IPs of its service using the following
mechanisms:

- Via the system's default DNS resolvers
- Via Google's DNS servers (`8.8.8.8` and `8.8.4.4`)
- Via Google's DNS-over-HTTPS service (https://developers.google.com/speed/public-dns/docs/doh)
- Via a file hosted on an ngrok-controlled S3 bucket

> Instead of using the system's default DNS resolvers, you may explicitly
> instruct the ngrok agent which DNS servers to use for resolution with the

Reviewer: ```suggestion
Instead of using the system's default DNS resolvers, you can configure
the DNS servers the ngrok agent uses for resolution with the
```
